platform/android/src/file_source.cpp

```diff
@@ -77,10 +77,19 @@ void FileSource::setResourceTransform(jni::JNIEnv& env, const jni::Object<FileSo
     }
 }
 
-void FileSource::setResourceCachePath(jni::JNIEnv& env, const jni::String& path) {
+void FileSource::setResourceCachePath(jni::JNIEnv& env, const jni::String& path,
+                                      const jni::Object<FileSource::ResourcesCachePathChangeCallback>& _callback) {
     std::string newPath = jni::Make<std::string>(env, path);
     mapbox::sqlite::setTempPath(newPath);
-    fileSource->setResourceCachePath(newPath + DATABASE_FILE);
+
+    auto global = jni::NewGlobal<jni::EnvAttachingDeleter>(env, _callback);
+    patchChangeCallback = std::make_unique<Actor<PathChangeCallback>>(*Scheduler::GetCurrent(),
```

Comment:
What happens if on java side developer would call `setResourcesCachePath` twice? First callback will never be called, right? Is that ok?
```java
setResourcesCachePath(path, new ResourcesCachePathChangeCallback {...} );
setResourcesCachePath(path, new ResourcesCachePathChangeCallback {...} );
```

Reply:
Good catch! This has been previously guarded in Java, but with recent streamlining of the process, I've removed that check. Let me adjust and add a test.